cmd/query/app/querysvc/query_service.go

```diff
@@ -38,12 +39,14 @@ const (
 type QueryServiceOptions struct {
 	ArchiveSpanReader spanstore.Reader
 	ArchiveSpanWriter spanstore.Writer
+	MetricsReader     metricsstore.Reader
```

Comment:
Why do we need this change? How does the existing v1 queryservice pass it and why would we need something different?

Reply:
Yes you are right, I will look into how its done in v1.